src/core/attributerange.cpp

```diff
@@ -27,23 +28,27 @@ AttributeRangePtr AttributeRange::parse(int attrId, const QString& attrName,
                                         const QString& attrRangeStr)
 {
     AttributeRangePtr vs;
+    QRegExp rx("([\\[{,]) +");  // replace consecutive spaces
```

Comment:
I think it does not catch the following case `a   ,   b`.
The current solution would fix it to `a   ,b` instead of `a,b`.

Reply:
Thanks for that! I was so fixated on treating spaces _after_ a special character, that I missed this (completely obvious) edge case.

Reply:
Thank you! I have a couple of questions:

* Should ``double  {1.1,1.2}`` be valid?

* How can I run the unit tests? Do I follow the steps described in the related [Qt tutorial](https://doc.qt.io/qt-5/qttestlib-tutorial1-example.html) ?

Reply:
> * Should `double  {1.1,1.2}` be valid?

Good question.
Yes, we could also accept the commands with leading/trailing whitespaces, i.e.,
`"              double    {     1.1     ,    1.2   }         "` could be the same as `"double{1.1,1.2}"`

> * How can I run the unit tests locally? Do I follow the steps described in the related [Qt tutorial](https://doc.qt.io/qt-5/qttestlib-tutorial1-example.html) ?

Yes, we use QTest. The only difference is that we use CMake instead of qmake.
In order to run the tests locally, you should enable the tests first, i.e., add the `-DTESTS=ON` flag to you cmake command, e.g., in the build folder
```bash
cmake.exe -DTESTS=ON -G "Visual Studio 15 2017 Win64" ../evoplex
```
then **compile** as you'd normally do. After that, to run all the tests, just call ` ctest.exe`.

Reply:
Ok I wrote the unit tests. I initially created a separate test ``tst_empty_spaces``, but grouping different attributes together (e.g. ``int[]``  and ``float{}``) didn't seem as intuitive as simply adding some additional cases to already existing tests.

I also noticed that while Travis passed, appveyor seems to be stuck and all the builds queued. Should I restart the build?

Reply:
> I also noticed that while Travis passed, appveyor seems to be stuck and all the builds queued. Should I restart the build?

It seems to be related to MSVC13 only. Don't worry about it.